platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TextField.kt

```diff
@@ -225,6 +233,26 @@ public fun TextField(
     }
 }
 
+@Composable
+private fun UndecoratedTextFieldDecorationBox(
+    innerTextField: @Composable () -> Unit,
+    placeholder: @Composable (() -> Unit)?,
+    textStyle: TextStyle,
+    placeholderTextColor: Color,
+) {
+    Box(propagateMinConstraints = true, contentAlignment = Alignment.CenterStart) {
+        innerTextField()
+
+        if (placeholder != null) {
```

Comment:
Ditto here, please make sure the placeholder is drawn behind the inner text field